File: frontend/src/app/models/utterance.ts
import { LabelClassification } from "./label-classification";

/**
 * Represents the utterance of a training dataset.
 */
export class Utterance {
    /**
     * Property identifier of the utterance.
     */
    code : string;

    /**
     * Question of the utterance.
     */
    utterance : string;

    /**
     *  Label classification for a utterance.
     */
    labelClassification : LabelClassification;

    /**
     * Constructor.
     *
     * @param code the property identifier of the utterance
     * @param utterance the utterance of the utterance
     * @param labelClassification the label classification for a utterance
     */
    constructor(code : string = "", utterance : string = "", labelClassification : LabelClassification = new LabelClassification()) {
        this.code = code;
        this.utterance = utterance;
        this.labelClassification = labelClassification;
    }

    /**
     * Returns a new {@link Utterance} object.
     *
     * @param Conversation a Conversation
     * @returns a new Conversation
     */
    public static getConversation(utterance: Utterance): Utterance {
        return new Utterance(
            utterance.code,
            utterance.utterance,
            utterance.labelClassification,
        );
    }
}

File: backend/models/Utterance.py

'''
Represents a utterance.

@author Eduardo Nascimento
'''
class Utterance():
    
    def __init__(self, code, utterance, label_classification):
        self.code = code  
        self.utterance = utterance
        self.label_classification = label_classification
    
    
    def __repr__(self):
        return '<Utterance - code: %r, utterance: %r, label: %s>' % (self.code, self.utterance, self.label_classification.name)
    
    def __str__(self):
        return '<Utterance - code: %r, utterance: %r, label: %s>' % (self.code, self.utterance, self.label_classification.name)

    def __dict__(self):
        return {'code': self.code, 'utterance': self.utterance, 'label_classification': self.label_classification.__dict__()}